packages/codegen-ui-react/lib/utils/format.ts

```diff
@@ -0,0 +1,1125 @@
+  if (validDate.toString() === invalidDateStr) {
+    return date;
+  }
+
+  const splitDate = date.split(/-|\+|Z/);
+
+  const year = splitDate[0];
+  const month = splitDate[1];
+  const day = splitDate[2];
+
+  // Remove leading zeroes
+  const truncatedMonth = month.replace(/^0+/, '');
+
+  switch (dateFormat) {
+    case 'locale':
+      return validDate.toLocaleDateString();
+    case 'YYYY.MM.DD':
+      return `${year}.${month}.${day}`;
+    case 'DD.MM.YYYY':
+      return `${day}.${month}.${year}`;
+    case 'MM/DD/YYYY':
+      return `${month}/${day}/${year}`;
+    case 'Mmm DD, YYYY':
+      return `${monthToShortMon[truncatedMonth]} ${day}, ${year}`;
+    default:
+      return date;
+  }
+}
+
+export function formatTime(time: string, timeFormat: TimeFormat['timeFormat']): string {
+  if (time === undefined || time === null) {
+    return time;
+  }
+
+  // AWSTime: hh:mm:ss.sss (24hr format - ISO 8601)
+  const splitTime = time.split(/:|Z/);
+
+  if (splitTime.length < 3) {
+    return time;
+  }
+
+  const validTime = new Date();
+  validTime.setHours(Number.parseInt(splitTime[0], 10));
+  validTime.setMinutes(Number.parseInt(splitTime[1], 10));
+
+  const splitSeconds = splitTime[2].split('.');
+  validTime.setSeconds(Number.parseInt(splitSeconds[0], 10), Number.parseInt(splitSeconds[1], 10));
+
+  if (validTime.toString() === invalidDateStr) {
+    return time;
+  }
+
+  switch (timeFormat) {
+    case 'locale':
+      return validTime.toLocaleTimeString();
+    case 'hours24':
+      return validTime.toLocaleTimeString('en-gb');
+    case 'hours12':
+      return validTime.toLocaleTimeString('en-us');
+    default:
+      return time;
+  }
+}
+
+export function formatDateTime(dateTimeStr: string, dateTimeFormat: DateTimeFormat['dateTimeFormat']): string {
+  if (dateTimeStr === undefined || dateTimeStr === null) {
+    return dateTimeStr;
+  }
+
+  // AWSTimestamp: millis before or after 1970-01-01-T00:00*Z*
+  // AWSDateTime: hh:mm:ss.sss (24hr format - ISO 8601)
+  const dateTime = /^\d+$/.test(dateTimeStr)
+    ? new Date(Number.parseInt(dateTimeStr, 10))
+    : new Date(Date.parse(dateTimeStr));
+
+  if (dateTime.toString() === invalidDateStr) {
+    return dateTimeStr;
+  }
+
+  if (dateTimeFormat === 'locale') {
+    return dateTime.toLocaleString();
+  }
+  const dateAndTime = dateTime.toISOString().split('T');
+  const date = formatDate(dateAndTime[0], dateTimeFormat.dateFormat);
+  const time = formatTime(dateAndTime[1], dateTimeFormat.timeFormat);
+
+  return `${date} - ${time}`;
+}
+
+export function format(value: string, formatterInput: FormatInputType) {
+  switch (formatterInput.type) {
+    case 'DateFormat':
+      return formatDate(value, formatterInput.format);
+    case 'DateTimeFormat':
+      return formatDateTime(value, formatterInput.format);
+    case 'TimeFormat':
+      return formatTime(value, formatterInput.format);
+    default:
+      return undefined;
```

Comment:
if we call the formatter on values that are undefined for format input - i think we should just return the value? or is the format util only being called on fields which need a format?

Reply:
this is the return never TS issue

Reply:
but eslint requires a default return

Reply:
The TS issue was because the last statement was inside the switch statement. We want to check for undefined for both value and formatterInput before it reaches the switch statement. If any of them is undefined, return the unformatted value

Reply:
I think returning the value inside the switch statement should be fine.